From 605472930a60281e2743b2d8d499884b79896568 Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Thu, 4 Apr 2019 15:11:16 -0600
Subject: [PATCH] makefile fixes

---
 makefile | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 1e32195..b7404cb 100644
--- a/makefile
+++ b/makefile
@@ -1,5 +1,12 @@
 LIBNAME = lpeg
 LUADIR = ../lua/
+VERSION ?= 1.0.2
+LUAVERSION ?= 5.3
+
+DESTDIR ?= ""
+PREFIX ?= $(DESTDIR)/usr
+LIBDIR = $(PREFIX)/lib64
+DATADIR = $(PREFIX)/share
 
 COPT = -O2 -DNDEBUG
 # COPT = -g
@@ -36,7 +43,7 @@ macosx:
 	$(MAKE) lpeg.so "DLLFLAGS = -bundle -undefined dynamic_lookup"
 
 lpeg.so: $(FILES)
-	env $(CC) $(DLLFLAGS) $(FILES) -o lpeg.so
+	env $(CC) $(DLLFLAGS)$(LDFLAGS) $(FILES) -o lpeg.so
 
 $(FILES): makefile
 
@@ -46,6 +53,13 @@ test: test.lua re.lua lpeg.so
 clean:
 	rm -f $(FILES) lpeg.so
 
+.PHONY: install
+install:
+	install -p -Dm 0755 lpeg.so $(LIBDIR)/lua/$(LUAVERSION)/lpeg.so.$(VERSION)
+	ln -sf lpeg.so.$(VERSION) $(LIBDIR)/lua/$(LUAVERSION)/lpeg.so
+	install -p -Dm 0644 re.lua $(DATADIR)/lua/$(LUAVERSION)/re.lua
+
+
 
 lpcap.o: lpcap.c lpcap.h lptypes.h
 lpcode.o: lpcode.c lptypes.h lpcode.h lptree.h lpvm.h lpcap.h
-- 
2.21.0

